MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/TrajectoryFilterWidget.py

```diff
@@ -0,0 +1,857 @@
+        """Re-renders the filter graph preview and updates the current filter settings when a setting is edited.
+
+        Parameters
+        ----------
+        key : str
+            The name of the edited setting.
+        value : any
+            The value of the edited setting.
+        """
+        # Update attribute
+        self._settings["attributes"].update({key: value})
+
+        # Check if attribute invokes change in how frequencies are passed to filter (single cutoff value or array of critical frequencies)
+        if self._settings["filter"] not in {"Notch", "Peak", "Comb"}:
+            if value in {"bandpass", "bandstop"}:
+                self.toggle_bound_frequencies()
+                self._settings["attributes"][
+                    "cutoff_freq"
+                ] = self.get_frequency_bounds()
+            elif value in {"lowpass", "highpass"}:
+                self.toggle_bound_frequencies(False)
+                self._settings["attributes"][
+                    "cutoff_freq"
+                ] = self.cutoff_freq_widget.value()
+            elif self.attenuation_type_widget.currentText() in {"bandpass", "bandstop"}:
+                self._settings["attributes"][
+                    "cutoff_freq"
+                ] = self.get_frequency_bounds()
+
+        # Re-render filter graph
+        self.render_canvas_assets()
+
+    def edit_preferences(self, key: str, value: any) -> None:
+        """Re-renders the filter according to display preferences.
+
+        Parameters
+        ----------
+        key : str
+            The name of the edited preference.
+        value : any
+            The value of the edited preference.
+        """
+        self._preferences.update({key: value})
+
+        # Load trajectory attenuation
+        if key == "show_attenuation" and not self._trajectory_power_spectrum:
+            self._trajectory_power_spectrum = power_spectrum(
+                self.find_configuration_property("trajectory"),
+                self.find_configuration_property("frames"),
+                self.find_configuration_property("projection"),
+                self.find_configuration_property("atom_selection"),
+                self.find_configuration_property("weights"),
+                self.find_configuration_property("instrument_resolution"),
+            )
+
+        self.render_canvas_assets()
+
+    def set_trajectory_power_spectrum(
+        self, filter: Filter
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Generate an appropriately resampled power spectrum for the input trajectory,
+        as well as the multiplicative attenuation effect of the designed filter.
+
+        Parameters:
+        ----------
+        filter: Filter
+            The Filter class for the designed filter
+
+        Returns:
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            Trajectory power spectrum and the attenuated power spectrum due to the designed filter response
```

Comment:
There are several issues with the numpydoc here, might want to run the numpydoc checker from the `numpydoc` package
```suggestion
        Parameters
        ----------
        filter : Filter
            The Filter class for the designed filter

        Returns
        -------
        Tuple[np.ndarray, np.ndarray]
            Trajectory power spectrum and the attenuated power spectrum due to the designed filter response
```